tests/rag_answers/test_evaluate.py

```diff
@@ -155,3 +157,78 @@ def test_evaluate_and_output_results_copes_with_empty_data(
     evaluate_and_output_results(mock_project_root, file_path, mock_evaluation_config)
 
     assert "There is no data to evaluate" in caplog.text
+
+
+def test_evaluate_and_output_results_writes_deepeval_test_run(
+    tmp_path,
+    mock_input_data,
+    mock_evaluation_config,
+    mock_run_deepeval_evaluation,
+    mocker,
+):
+    class DummyTestRun:
+        def __init__(self) -> None:
+            self.call_kwargs: dict[str, bool] | None = None
+
+        def model_dump(self, *, by_alias: bool, exclude_none: bool) -> dict[str, str]:
+            self.call_kwargs = {"by_alias": by_alias, "exclude_none": exclude_none}
+            return {"id": "test-run-123"}
+
+    test_run = DummyTestRun()
```

Comment:
I think this object should be related to the class it is mocking, TestRun, as there isn't anything here that ties it the interface that provides (kinda like instance double in RSpec).

I'm not sure exactly how we do this with Python mocking but from a quick google I think it could be autospeccing